python/pyspark/tests.py

```diff
@@ -740,6 +739,27 @@ def test_multiple_python_java_RDD_conversions(self):
         converted_rdd = RDD(data_python_rdd, self.sc)
         self.assertEqual(2, converted_rdd.count())
 
+    def test_narrow_dependency_in_join(self):
+        rdd = self.sc.parallelize(range(10)).map(lambda x: (x, x))
```

Comment:
This test is only for correctness, I will add more check for narrow dependency base one the Python progress API (#3027)